ShapeVariationAnalyzer/ShapeVariationAnalyzer.py

```diff
@@ -2246,8 +2246,23 @@ def addColorMap(self, table, dictVTKFiles):
                     polyData = append.GetOutput()
 
                     vtkFile = os.path.splitext(vtkFile)[0]+'.vtk'
+                elif os.path.basename(vtkFile).endswith(".srep.json"):
+                    storage = slicer.mrmlScene.AddNewNodeByClass('vtkMRMLSRepStorageNode')
+                    storage.SetFileName(vtkFile)
+                    try:
+                        srep = storage.CreateSRepNode("name")
+                    except:
+                        slicer.mrmlScene.RemoveNode(storage)
+                        return False
+                    try:
+                        exportProps = slicer.vtkSRepExportPolyDataProperties()
+                        logic = slicer.vtkSlicerSRepLogic()
+                        polyData = logic.ExportSRepToPolyData(srep.GetSRep(), exportProps)
```

Comment:
I don't think `convertSrepToPolyData` would be a good name, since converting to a polydata in memory and writing the polydata to a file are two different things. Maybe it would help simplify some things if `ExportSRepToPolyData` took a default nullptr for the exportProps and used a default constructed one if it was nullptr.

Regarding the function taking a path to a `.vtk`, you are asking for something like `WriteSRepAsPolyData(srep, filepath)`?

Reply:
The other thing that would simplify things would be the addition of something like a TempNode or ScopedNode context manager in slicer/util.py

Would be used like this
```
with TempNode(storage.CreateSRepNode("name")) as srep:
    exportProps = slicer.vtkSRepExportPolyDataProperties()
    logic = slicer.vtkSlicerSRepLogic()
    polyData = logic.ExportSRepToPolyData(srep.GetSRep(), exportProps)
```

This would help reduce the all the `try/finally`s

Reply:
If you do not object, I will leave this as is, but I do think a `TempNode`-esque class would be beneficial and should be discussed.